src/commands/diff.rs

```diff
@@ -207,40 +261,53 @@ fn diff(
     let mut item1 = tree_streamer1.next().transpose()?;
     let mut item2 = tree_streamer2.next().transpose()?;
 
+    let mut diff_statistics = DiffStatistics::default();
+
     loop {
         match (&item1, &item2) {
             (None, None) => break,
             (Some(i1), None) => {
                 println!("-    {:?}", i1.0);
+                diff_statistics.removed_node(&i1.1.node_type);
                 item1 = tree_streamer1.next().transpose()?;
             }
             (None, Some(i2)) => {
                 println!("+    {:?}", i2.0);
+                diff_statistics.added_node(&i2.1.node_type);
                 item2 = tree_streamer2.next().transpose()?;
             }
             (Some(i1), Some(i2)) if i1.0 < i2.0 => {
                 println!("-    {:?}", i1.0);
+                diff_statistics.removed_node(&i1.1.node_type);
                 item1 = tree_streamer1.next().transpose()?;
             }
             (Some(i1), Some(i2)) if i1.0 > i2.0 => {
                 println!("+    {:?}", i2.0);
+                diff_statistics.added_node(&i2.1.node_type);
                 item2 = tree_streamer2.next().transpose()?;
             }
             (Some(i1), Some(i2)) => {
                 let path = &i1.0;
                 let node1 = &i1.1;
                 let node2 = &i2.1;
                 match &node1.node_type {
-                    tpe if tpe != &node2.node_type => println!("T    {path:?}"), // type was changed
+                    tpe if tpe != &node2.node_type => {
+                        // type was changed
+                        println!("T    {path:?}");
+                        diff_statistics.changed_file();
+                    }
                     NodeType::File if !no_content && !file_identical(path, node1, node2)? => {
                         println!("M    {path:?}");
+                        diff_statistics.changed_file();
                     }
                     NodeType::File if metadata && node1.meta != node2.meta => {
                         println!("U    {path:?}");
+                        diff_statistics.changed_file();
```

Comment:
also here. This is changed metadata, e.g. the the time has changed.

Reply:
Same, I added a `Metadata` change type for that